src/main/java/org/commonwl/view/git/GitDetails.java

```diff
@@ -242,4 +275,67 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(repoUrl, branch, path, packedId);
     }
+
+    public String getLicense(Path workTree) {
+        switch (getType()) {
+            case GITHUB:
+                String githubApiUrl = "https://api.github.com/repos/" +
+                        normaliseUrl(repoUrl).replace("github.com/", "").replace(".git", "") +
+                        "/license";
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Contacting GitHub api at url " + githubApiUrl);
+                }
+                RestTemplate githubRestTemplate = new RestTemplate();
+                ObjectNode githubJsonLicense = githubRestTemplate.getForObject(githubApiUrl, ObjectNode.class);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Received response:\n\n" + githubJsonLicense.toPrettyString());
+                }
+                if (githubJsonLicense.has("license") &&
+                        !"other".equals(githubJsonLicense.get("license").get("key").asText())) {
+                    return SPDX_PREFIX + githubJsonLicense.get("license").get("spdx_id").asText();
+                } else if (githubJsonLicense.has("download_url")) {
+                    return githubJsonLicense.get("download_url").asText();
+                } else {
+                    return null;
+                }
+            case GITLAB:
+                String gitlabApiUrl = "https://gitlab.com/api/v4/projects/" +
```

Comment:
Would be nice to link to the docs in a comment

https://docs.gitlab.com/ee/api/projects.html#get-single-project mentions the license field, but it doesn't document the response. Do they have an OpenAPI doc site or similar?